### DISCUSSION TITLE
Multiple for loops are not supported to deduce one list

### DISCUSSION THREAD

**Discussion by BrainPicker-L** [cue-lang/cue#1651] [Q&A]
### What version of CUE are you using (`cue version`)?

<pre>
$ cue version latest

</pre>

### What did you do?
I want to deduce 1 list through multiple for loops. but it's not work
```
	Nodes: [ for n in preOrchestration.Nodes {
		[ for comp in plan.productComponentSet {
			if n.Name == comp.Name {
				[ for ins in comp.InstanceSet {
					Name: ins.InstanceName
				}
				]
			}
		}
		]
	}
	]
```
Only 1 list can be deduced by 1 for loop，e.g.
```
listOfStructs: [ for p, n in nums {
	pos: p
	val: n
}]
```

### What did you expect to see?
I would like to get an example of multiple for loops deriving 1 list

**Comment by myitcv** [comment]
Hi @BrainPicker-L welcome to the CUE project!

I've converted your issue to be a discussion for now, because it's not clear right now that there is a bug/feature request. 

Please can you provide a full reproducer of the issue you're seeing? It's a little hard to understand exactly what you're expecting here.

Thanks

**Comment by verdverm** [comment]
You can use `list.FlattenN([...[...]], D)` to, well, flatten lists D times

So `NodeList: list.FlattenN(Nodes, 1)`
